sql/sql_view.cc

```diff
@@ -453,38 +453,81 @@ bool mysql_create_view(THD *thd, TABLE_LIST *views,
     goto err;
   }
 
-  /* Implicitly add versioning fields if needed */
+  /* System Versioning begin */
+  for (TABLE_LIST *table= tables; table; table= table->next_local)
   {
-    TABLE_LIST *tl = tables;
-    while (tl && tl->is_view())
-      tl = tl->view->select_lex.table_list.first;
-    if (tl && tl->table)
+    DBUG_ASSERT(!table->is_view() || table->view);
+    // Any versioned table in VIEW will add `FOR SYSTEM_TIME ALL` + WHERE:
+    // if there are at least one versioned table then VIEW will contain FOR_SYSTEM_TIME_ALL
+    // (because it is in fact LEX used to parse its SELECT).
+    if (table->is_view() && table->view->vers_conditions == FOR_SYSTEM_TIME_ALL)
     {
-      TABLE_SHARE *s= tl->table->s;
-      if (s->versioned)
+      my_error(ER_VERS_VIEW_PROHIBITED, MYF(0), table->table_name);
+      res= true;
+      goto err;
+    }
+
+    if (!table->table)
+      continue;
+
+    /* Implicitly add versioning fields if needed */
+    if (table->table->versioned())
+    {
+      Item *item;
+      List_iterator_fast<Item> it(select_lex->item_list);
+      bool add_start= true;
```

Comment:
Try this test case. `SELECTS` returns nothing here, while it should return 1 row.
```sql
create table t (a int) with system versioning;
insert into t values (1);
create view vt as select a, sys_trx_end, sys_trx_start from t;
select *, sys_trx_start, sys_trx_end from vt;
```

Our code relies on particular vers field position and order. As I understand, standard requires it too.